### ISSUE TITLE
Replace al the 'local' aria-live regions with @wordpress/a11y speak

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#79530]
### Description

Discovered while investigating https://github.com/WordPress/gutenberg/issues/79526
Cc @joedolson for visibility

Currently, there are a few usages of aria-live regions in the codebase that are less than ideal for various reasons:
- They are inconsistent, as in they use important attributes for aria live regions in an inconsistent way or they don't use them at all.
- These aria live regions may be dynamically 'mounted' (injected in the DOM) together with the component they belong to: this is a no-go because aria-live regions **must** be in the DOM on page load and stay there so that screen reader can monitor them.

For example, this kind of implementation:

```
<div aria-live="polite">{ visibleMessage }</div>
```

Is not OK. It misses a few necessary attributes to make the live region work reliably. Also, the passed `visibleMessage` is the `ValidityIndicator` component, a paragraph with an icon. That's not ideal. Live regions work best when they are passed a text string. HTML should be avoided.

There is no reason to use these 'local' aria live regions. WordPress uses since several years a dedicated package (I think it is available also as a module) for this purpose, which has been tested for years and it's a consistent, centralized, implementation.

Some of the occurrences of 'local' aria live regions I found are in these files (there may be others):

- `packages/components/src/validated-form-controls/control-with-error.tsx`
- `packages/dataviews/src/components/dataform-controls/date.tsx`
- `packages/fields/src/components/media-edit/index.tsx`
- `packages/media-editor/src/image-editor/react/components/cropper.tsx`

### Step-by-step reproduction instructions

N/A

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by talldan** [comment]
cc @ramonjd @andrewserong for the media/cropper related usage.

**Comment by ramonjd** [comment]
Looks like there's a PR up already 🎉 

Is https://github.com/WordPress/gutenberg/pull/79600 looking like the right direction?

**Comment by talldan** [comment]
I wasn't sure about it because the screenshots for the media editor looked a bit unusual, but now I notice they look like they're from the storybook.

**Comment by joedolson** [comment]
@ramonjd Yes, the direction looks right.
